plugins/metrics/message.go

```diff
@@ -70,9 +106,33 @@ var (
 	// current number of missing messages in missingMessageStorage
 	missingMessageCountDB atomic.Uint64
 
+	// current number of finalized messages
+	finalizedMessageCount = make(map[MessageType]uint64)
+
+	// protect map from concurrent read/write.
+	finalizedMessageCountMutex syncutils.RWMutex
+
+	// total time it took all messages to finalize. unit is milliseconds!
+	messageFinalizationTotalTime = make(map[MessageType]uint64)
+
+	// protect map from concurrent read/write.
+	messageFinalizationTotalTimeMutex syncutils.RWMutex
+
+	// current number of confirmed  branches
+	confirmedBranchCount atomic.Uint64
+
+	// total time it took all branches to finalize. unit is milliseconds!
```

Comment:
same.. use nanosec or time.Duration

Reply:
Nanoseconds could overflow for uint64 and for metrics we don't need that much precision anyway.